src/libcore/iter/adapters/zip.rs

```diff
@@ -36,6 +36,22 @@ impl<A: Iterator, B: Iterator> Zip<A, B> {
     }
 }
 
+impl<A: DoubleEndedIterator, B: DoubleEndedIterator> Zip<A, B>
+where
+    A: DoubleEndedIterator + ExactSizeIterator,
+    B: DoubleEndedIterator + ExactSizeIterator,
```

Comment:
This isn't really specialization, right? That is, the `super_nth_back` method appears to only be available for `Zip` of two double-ended iterators, and not in any other cases...

Reply:
`next_back` is also specialized for `DoubleEndedIterator + ExactSizeIterator` so I thought it would be right to do the same here. But I may have missed something, do you have any better approach?

Reply:
I see, I was confused, ignore me.